openedx/core/djangoapps/agreements/serializers.py

```diff
@@ -17,4 +17,17 @@ class IntegritySignatureSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = IntegritySignature()
-        fields = ('username', 'course_id', 'created_at')
+
+
+class LTIPIISignatureSerializer(serializers.ModelSerializer):
+    """
+    Serializer for LTIPIISignature model
+    """
+    username = serializers.CharField(source='user.username')
+    course_id = CourseKeyField(source='course_key')
+    lti_tools = serializers.JSONField()
+    created_at = serializers.DateTimeField(source='created')
+
+    class Meta:
+        model = LTIPIISignature()
+        fields = ('username', 'course_id', 'lti_tools', 'created_at')
```

Comment:
One of the nice benefits of the `ModelSerializer` class over the `Serializer` class is that it will automatically create these serializer fields for you based on the model definition. You can read more about Django Rest Framework (DRF) `ModelSerializers` [here](https://www.django-rest-framework.org/api-guide/serializers/#modelserializer).

The reason we define `username` and `course_id` serializer fields explicitly is because, as you'll notice, those are not fields of the `LTIPIISignature` model. So, we need to define the serializer fields explicitly because DRF cannot automatically create these fields for us under the hood. It doesn't know what these fields are. The reason we define a `username` field and don't use what would be generated for the `user` model field is because we want the frontend to send us the user's username and not their database ID.

However, `lti_tools` _is_ on the model. Therefore, we do not have to include an explicit serializer field for it. In the below code snippet, I've removed the `lti_tools` serializer field from the serializer and show that it's correctly generated by DRF for us. I had to remove references to `created_at` to get this to work, by the way, due to an issue I mention in another comment.

```
Python 3.8.10 (default, May 26 2023, 14:05:08)
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from openedx.core.djangoapps.agreements.serializers import LTIPIISignatureSerializer
>>> serializer = LTIPIISignatureSerializer()
>>> print(repr(serializer))
LTIPIISignatureSerializer():
    username = CharField(source='user.username')
    course_id = CourseKeyField(source='course_key')
    lti_tools = JSONField(decoder=None, encoder=None, style={'base_template': 'textarea.html'})
```